### ISSUE TITLE
[Enhancement] Add ability to pass function as value inside of `excludes` array.

### ISSUE THREAD

**Issue by TheLarkInn** [NekR/offline-plugin#236]
Currently if you have a unique scenario where `minimatch` cannot find a complex or questionable path, there is no fallback for filtering assets that are allowed to go into the sw.js and appcache manifest. 

With that in mind I'd like to propose a two part enhancement to offline-plugin. 

## Phase 1: [Partial Implementation]: Allow predicate function to be passed as a value inside of the used array. 

This may look like `excludes: [(asset) => /\.foo$/.test(asset), ..., ..., ...]`

## Phase 2: Implement the same condition matching system for resource (path names) that webpack uses across all of its plugin values. (See RuleSet.js#normalizeConditions, applyMatchers)

**Comment by NekR** [comment]
I think partial implementation somehow should go into v4, even though I planned to not really any version of it except for bug fixing. We have a problem of merging the same code into both v4 and v5.

Full implementation definitely goes to v5.

**Comment by NekR** [comment]
Also, I'm not 100% sure about the "full implementation". It's a bit different in `offline-plugin` than webpack itself, so maybe we'll end up using just a simple wrapper for all matches which will be accepting glob/regexp/functions.

**Comment by TheLarkInn** [comment]
Correct, I would think of the condition in this case as unique to the "exclude" property, where essentially a string (abs path), regex, function() {}, and I suppose if you wanted to make it recursive you could but maybe that is where the line is drawn.

**Comment by NekR** [comment]
@TheLarkInn do you mind looking into this https://github.com/NekR/offline-plugin/pull/242? Seems like a thing related to this, so would be good to have a solution which fits both cases.
